### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Nejjari001**
## Loom Recording  [video](https://www.loom.com/share/5062fe035e884a75bc881e445e98e729)
- Initialise React application using create-react-app
- Fetch data from CoinApp API using AXIOS library
- Filter stored data by their Rank
- Display the data along with the crypto name
- When the user clicks on the crypto name, the application navigates to the details page.
- Retrieve data from the API to show detailed data in the category.
- When the user clicks on the "Back" button, the user navigates to the home page.
- Write unit tests for pure functions in your Redux code.
- Write integration tests using the React Testing Library.

### CODE REVIEW

File: src/__testing__/App.test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { render } from '@testing-library/react';
+import { BrowserRouter } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import App from '../App';
+import Store from '../Redux/ConfigureStore';
+
+const MockApp = () => (
```
Comment (julie-ify) [inline src/__testing__/App.test.js:7]: - [ ] One of your tests is failing. It is in the `src/App.test.js` file. Kindly fix that.

  ddudu

File: src/App.js [hunk 2]
```diff
@@ -1,24 +1,32 @@
...
+        path="/"
+        element={(
+          <>
+            <NavBar />
+            <Crytpos />
+          </>
+        )}
+      />
+      <Route
+        path="/crypto-details"
```
Comment (julie-ify) [inline src/App.js]: - [ ] All your category routes have the same `crypto-details` as their route, but EACH item is required to have its own unique route. Then you can fetch the item from the store or API using the unique identifier.

  For example, you should have something like this:
  ```
  path="crypto-details/:id"
  ```
  or
  ```
  path=":id"
  ```

  Suggestion: You can consider looking at the `useParams` from the `react-router-dom` package for an idea of how to do that.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Nejjari001,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 3]
```diff
@@ -1,70 +1,174 @@
-# Getting Started with Create React App
+<div align="center">
```
Comment (DammyShittu) [inline README.md:1]: - [ ] Your test is still failing. As you can see from the screenshot below and as highlighted by the previous reviewer, the failing test is in the `src/App.test.js` file. Kindly fix that.

  image

File: src/App.js (hunk 2 above)
Comment (DammyShittu) [inline src/App.js]: - [ ] Kindly implement the unique routing as requested by the previous reviewer. Each crypto token on the home page should have its own unique route. Kindly observe the earlier request for a clue on how to go about this.

  You can consider using the useParams module from the react-router package as suggested.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Nejjari001,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Components/Crytpos.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    navigate(
+      '/crypto-details',
+      {
+        state: {
+          cryptos: cryptoObj,
+        },
+      },
+
+    );
+  };
```
Comment (ShadyShawkat) [inline src/Components/Crytpos.js]: - [ ] Nice job changing the URL as the previous reviewers suggested. However, you forgot to change the URL you navigate to when the user clicks on a "crypto" item which produced the following warning in the console. Please make sure to test your app with every change and see if everything works as intended.

image

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @Nejjari001, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

Image

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Components/Crytpos.js (hunk 4 above)
Comment (Nejjari001) [reply src/Components/Crytpos.js]: ## Hello @ShadyShawkat   👋🏼 
 Thanks for taking out time to review this PR. The required and optional changes have been duly noted and necessary changes will be implemented.

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Nejjari001👋, 

### You did a marvelous job on the React capstone project and this is one of the best projects I have seen so far!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- All tests are passed. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**